databases/pxlib/Portfile

```diff
@@ -28,22 +27,30 @@ checksums           rmd160  1bf8b7e14c92f47058aebd1a675b34a76b720660 \
                     sha256  4d0a84a36d46a81ff8c629c4a00de3e7b6e7327d5d56c0a2ae44c1ea715ae63f \
                     size    532895
 
-depends_build       port:pkgconfig \
-                    port:intltool \
-                    port:autoconf \
-                    port:automake \
-                    port:libtool
-
-depends_lib         port:libgsf \
-                    port:libiconv
-
 # reconfigure using autogen.sh for intltool 0.51 compatibility
 
 patchfiles          patch-autogen.sh.diff \
                     patch-configure.ac.diff
 
 configure.cmd       ./autogen.sh && ./configure
 
-configure.args      --disable-silent-rules \
+depends_build-append \
+                    port:autoconf \
+                    port:automake \
+                    port:gettext \
+                    port:intltool \
+                    port:libtool \
+                    port:pkgconfig
+
+depends_lib-append  port:libgsf \
+                    port:libiconv
+
+configure.args-append \
+                    --disable-silent-rules \
```

Comment:
Most of this has nothing to do with building with recent clang.

Reply:
It can be seen from the history that initially I just added the flag, but that was not sufficient: https://github.com/macports/macports-ports/compare/d667bf076d8f2dc5ab31cff6c1601cc6aaeb299f..8468e404b346750a6b98c9b2e5916cf0f5172630

Reply:
> If I remember correctly, when a custom command is used for autoreconf, build dependencies should go after it and not prior to.

That is true, but a custom command for autoreconf has not been used here.

Setting `use_autoreconf yes` adds the autoconf, automake, and libtool build dependencies. If you then override `autoreconf.cmd`, MacPorts removes those dependencies again. Then you have to add them back in manually if the custom command you're running (often `./autogen.sh` or `./bootstrap`) still needs those dependencies.

In this port, `use_autoreconf yes` and `autoreconf.cmd` are not being used so none of that applies; it will work regardless of where you specify the dependencies, although the usual flow of a Portfile is to write dependencies before the directives relating to the patch, configure, build, and destroot phases.

Reply:
> It can be seen from the history that initially I just added the flag, but that was not sufficient: https://github.com/macports/macports-ports/compare/d667bf076d8f2dc5ab31cff6c1601cc6aaeb299f..8468e404b346750a6b98c9b2e5916cf0f5172630

pxlib previously failed to build in CI because:

```
configure.ac:88: error: possibly undefined macro: AM_ICONV
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
configure:3361: error: possibly undefined macro: AM_NLS
```

The first error probably would be solved by adding a build dependency on the libiconv port and the second would be solved by adding a build dependency on the gettext port—which you did in this PR, and gettext depends on libiconv, so that's probably fine.

So the previous build failure had nothing to do with the presence or absence of a compiler flag—it never got to the compilation stage.

Reply:
@ryandesign Thank you, Ryan. I will update the PR today.